kasa/smartprotocol.py

```diff
@@ -101,51 +102,79 @@ async def _query(self, request: Union[str, Dict], retry_count: int = 3) -> Dict:
         # make mypy happy, this should never be reached..
         raise SmartDeviceException("Query reached somehow to unreachable")
 
+    async def _execute_multiple_query(self, request: Dict, retry_count: int) -> Dict:
+        debug_enabled = _LOGGER.isEnabledFor(logging.DEBUG)
+        multi_result: Dict[str, Any] = {}
+        smart_method = "multipleRequest"
+        requests = [
+            {"method": method, "params": params} for method, params in request.items()
+        ]
+
+        end = len(requests)
+        # Break the requests down as there seems to be a size limit
+        step = self.MULTI_REQUEST_BATCH_SIZE
```

Comment:
Instead of relying on a class-level variable for this, it'd be better to add a parameter for the method itself. That'll make it easier to write a test for the batching behavior.

Reply:
Rather than add a parameter which would break the `BaseProtocol` interface or require updates to all protocols I added this to the `DeviceConfig`.